editor/src/core/commenting/comment-types.tsx

```diff
@@ -0,0 +1,132 @@
+import type { CanvasPoint, LocalPoint } from '../shared/math-utils'
+import { canvasPoint, localPoint } from '../shared/math-utils'
+import { assertNever } from '../shared/utils'
+
+export type UtopiaThreadMetadata = CanvasThreadMetadata | SceneThreadMetadata
+
+type BaseThreadMetadata = {
+  position: CanvasPoint
+  resolved: boolean
+}
+
+export type CanvasThreadMetadata = BaseThreadMetadata & {
+  type: 'canvas'
+}
+
+export type SceneThreadMetadata = BaseThreadMetadata & {
+  type: 'scene'
+  sceneId: string
+  scenePosition: LocalPoint
+  remixLocationRoute?: string
+}
+
+export function canvasThreadMetadata(
+  position: CanvasPoint,
+  resolved: boolean = false,
+): CanvasThreadMetadata {
+  return {
+    type: 'canvas',
+    position: position,
+    resolved: resolved,
+  }
+}
+
+export function sceneThreadMetadata(
+  position: CanvasPoint,
+  sceneId: string,
+  scenePosition: LocalPoint,
+  remixLocationRoute?: string,
+  resolved: boolean = false,
+): SceneThreadMetadata {
+  return {
+    type: 'scene',
+    position: position,
+    sceneId: sceneId,
+    scenePosition: scenePosition,
+    remixLocationRoute: remixLocationRoute,
+    resolved: resolved,
+  }
+}
+
+export function isSceneThreadMetadata(
+  metadata: UtopiaThreadMetadata,
+): metadata is SceneThreadMetadata {
+  return metadata.type === 'scene'
+}
+
+export function isCanvasThreadMetadata(
+  metadata: UtopiaThreadMetadata,
+): metadata is CanvasThreadMetadata {
+  return metadata.type === 'canvas'
+}
+
+export function liveblocksThreadMetadataToUtopia(metadata: ThreadMetadata): UtopiaThreadMetadata {
+  if (metadata.sceneId != null && metadata.sceneX != null && metadata.sceneY != null) {
+    return {
+      type: 'scene',
+      position: canvasPoint(metadata),
+      sceneId: metadata.sceneId,
+      scenePosition: localPoint({ x: metadata.sceneX, y: metadata.sceneY }),
+      resolved: metadata.resolved,
+    }
+  } else {
+    return {
+      type: 'canvas',
+      position: canvasPoint(metadata),
+      resolved: metadata.resolved,
+    }
+  }
+}
+
+export function utopiaThreadMetadataToLiveblocks(metadata: UtopiaThreadMetadata): ThreadMetadata {
+  switch (metadata.type) {
+    case 'canvas':
+      return {
+        x: metadata.position.x,
+        y: metadata.position.y,
+        resolved: metadata.resolved,
+      }
+    case 'scene':
+      return {
+        x: metadata.position.x,
+        y: metadata.position.y,
+        sceneId: metadata.sceneId,
+        sceneX: metadata.scenePosition.x,
+        sceneY: metadata.scenePosition.y,
+        resolved: metadata.resolved,
+        remixLocationRoute: metadata.remixLocationRoute,
+      }
+    default:
+      assertNever(metadata)
+  }
+}
+
+type PartialNullable<T> = {
+  [P in keyof T]?: T[P] | null | undefined
+}
+
+export function utopiaThreadMetadataToLiveblocksPartial(
+  metadata: PartialNullable<UtopiaThreadMetadata>,
+): PartialNullable<ThreadMetadata> {
+  if (metadata.type === 'scene') {
```

Comment:
I would go with a switch/case here so that `assertNever` can throw an error if we ever add a new comment type

Reply:
assertNevert doesn't work here because of the Partial type